### ISSUE TITLE
How to disable `rule.exported` for consts?

### ISSUE THREAD

**Issue by Potherca** [mgechev/revive#1045]
## TL;DR:

I am aware that the goal of this project is no longer to be a 100% drop-in replacement for golint (which I think is fair) but is there a way for me to disable the `exported` rule but only for consts?

## Full Details

At [pipeline-components](https://pipeline-components.dev/) we have a component for [go-lint](https://gitlab.com/pipeline-components/go-lint). Although it is not one of our larger components<sup>\[1]</sup>, I'm thinking of prolonging support by replacing `golint` with revive.

However, the current claim that the default behaviour is the same as `golint` does not appear to be 100% accurate.

To be specific, I am seeing the following warnings from revive that are not there with golint:

1. "should have a package comment"
2. "this block is empty, you can remove it"
3. "comment on exported const [..] should be of the form [..]"

The first two I can resolve by tweaking the default config,<sup>\[2]</sup>, which is fine for our purposes (as golint does not have a config, so we can just add one for revive).

But I have not yet been able to resolve the `comment on exported const [..] should be of the form [..]`. Please note, this is explicitly about exported "const", not other exported types.

How could this be achieved?

---

<sup>\[1]</sup> Currently it only has +/- 40K pulls on DockerHub ![][1]<br>
<details><summary><sup>[2]</sup> Click to see config</summary>

```toml
ignoreGeneratedHeader = false
severity = "warning"
confidence = 0.8
errorCode = 0
warningCode = 0

[rule.blank-imports]
[rule.context-as-argument]
[rule.context-keys-type]
[rule.dot-imports]
[rule.empty-block]
    Disabled = true
[rule.errorf]
[rule.error-naming]
[rule.error-return]
[rule.error-strings]
[rule.exported]
[rule.increment-decrement]
[rule.indent-error-flow]
[rule.package-comments]
     Disabled = true
[rule.range]
[rule.receiver-naming]
[rule.redefines-builtin-id]
[rule.superfluous-else]
[rule.time-naming]
[rule.unexported-return]
[rule.unreachable-code]
[rule.unused-parameter]
[rule.var-declaration]
[rule.var-naming]
```

</details>

[1]: https://img.shields.io/docker/pulls/pipelinecomponents/go-lint.png

**Comment by chavacava** [comment]
Hi @Potherca, I'm not sure to understand your request on linting comments on public constants.
For the file `foo.go` 
```go
package main

// this is not ok
const AConstant = 10
```

When linting it with [`golint`](https://github.com/golang/lint/tree/master) I get:
```
$ golint .
foo.go:3:1: comment on exported const AConstant should be of the form "AConstant ..."
```
and when linting with `revive` I get the same failure:
```
$ revive .
foo.go:3:1: comment on exported const AConstant should be of the form "AConstant ..."
```
Thus it seems to me that for the case of public constants comments, `revive` is iso-functional to `golint`. I'm missing something?

**Comment by mfederowicz** [comment]
Hi @chavacava i think @Potherca wanted to switch off only some warnings (related with const), but there are no option for that. So I try to add this feature in PR #1047 think about it :)

**Comment by chavacava** [comment]
Hi @mfederowicz, as mentioned in the OP, two of the three warnings can be avoided by tweaking the configuration. The third one,  "comment on exported const [..] should be of the form [..]" can not be disabled but my understanding is that in what concerns to that particular warning `revive` behaves as `golint` does thus no need to avoid it. Then, if I'm not missing something, there is no need to modify `revive`.

**Comment by mfederowicz** [comment]
ok @chavacava I made my PR because I think that should be option to switch of selected warnings, but this is my opinion. On the other way fixing code  for consts  (in scaned project) shouldnt be complicated process :)

**Comment by Potherca** [comment]
@chavacava Thank you for taking the time to look at this.

To reproduce what I am seeing:

```go
package main

const (
	// this is not ok
	AConstant = 10

	Bconstant = 11 // Some comment here
)
```

Running revive and golint side-by-side[^1] shows:

[^1]: Using `diff <(~/go/bin/revive ./... | sort) <(docker run --rm -v "${PWD}:/code/" pipelinecomponents/go-lint ./... | sort)`

```diff
1,2d0
< 
< main.go:1:1: should have a package comment
4c2
< main.go:7:17: comment on exported const Bconstant should be of the form "Bconstant ..."
---
> main.go:7:2: exported const Bconstant should have comment (or a comment on this block) or be unexported
```

So it looks like the comment behind the const is the cause of the difference between golint and revive...

**Comment by chavacava** [comment]
@Potherca, thanks for the details. I'll further investigate these kind of cases.

**Comment by chavacava** [comment]
Hi @Potherca could you please check if the PR #1051 solves the problem?

**Comment by Potherca** [comment]
I can confirm that #1051 resolves all issues I have. Steps taken:

1. Build the patched `revive`
    ```sh
    git -C /tmp clone https://github.com/chavacava/revive.git
    cd /tmp/revive
    go build main.go
    ```
 2. Run the patched version and golint on my project and compare the results:
    ```sh
    diff <(//tmp/revive/main -config default.toml ./... | sort) <(~/go/bin/golint ./... | sort)
    ```
